### PR TITLE
Extend the plugin with Spout Output

### LINKED ISSUE #14: Bounty on spout output
Dear friends. We like to support the development of a spout output feature with a 500eur bounty.

We are a unreal developer team. Last project was "ColoryVR" which is released on steam.

Any response would be helpful.

Best regards, ed.
www.lucidlayers.com

### PR DESCRIPTION
**Pull request by mzlt**
* Initialise a DX texture from the raw OBS output buffer
* Use base Spout API to copy intitialised texture to shared memory
* Create UI for controlling Spout output

Also:
* Update CMAKE config to simplify build
* Move spout `deps` directory inside plugin
* Revert to Spout 2.006 due to Spout API bug
* Target OBS v 26.1.1

Closes #14  #11

### CODE REVIEW

File: source/win-spout.cpp [hunk 1]
```diff
@@ -8,31 +8,61 @@
...
+	obs_data_t* settings = obs_data_create();
+	win_spout_out = obs_output_create("spout_output", "OBS Spout Output", settings, NULL);
+	obs_data_release(settings);
+
+	QAction* menu_action =
+		(QAction*)obs_frontend_add_tools_menu_qaction(
+		obs_module_text("toolslabel"));
+
+	obs_frontend_push_ui_translation(obs_module_get_string);
+	spout_output_settings = new win_spout_output_settings(main_window);
```
Comment (campbellwmorgan) [inline source/win-spout.cpp:61]: Does this not need to be deleted on unload?

**Review by campbellwmorgan — COMMENTED** [review]
Great work - 2 questions that might need answering!

File: source/win-spout-output.cpp [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+	context->output_started = false;
+	context->reset_info();
+
+	win_spout_output_update(context, settings);
+
+	if (!init_spout(context))
+	{
+		blog(LOG_ERROR, "Failed to init spout!");
+		deinit_spout(context);
+		return false;
```
[the remark this answers is not in the document]
Comment (mzlt) [reply source/win-spout-output.cpp:138]: 100% this is an oversite, it should return `NULL`. Looking at it now I would probably like to reorder this function a little bit as well for readability

File: source/win-spout.cpp (hunk 1 above)
Comment (mzlt) [reply source/win-spout.cpp:61]: Good catch, we should add that in!

File: source/win-spout-output.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (mzlt) [reply source/win-spout-output.cpp]: Fixed now

File: source/win-spout.cpp (hunk 1 above)
Comment (campbellwmorgan) [reply source/win-spout.cpp:61]: appears that calling `delete` on this crashes OBS, but without it OBS closes correctly even opening multiple versions of obs and closing, so assuming  that this is not a memory leak we need to worry about

**Pull request merged**